client/src/client/__init__.py

```diff
@@ -115,20 +116,36 @@ def __init__(self) -> None:
                     assert mnemonic and new_passphrase is not None
 
                 self.password = self.startup.get_key(user_passphrase)
-                self.priv_key, self.pub_key = self.startup.load_key_pair(
-                    self.password
-                )
-                self.account = self.startup.load_crypto_account(
-                    user_passphrase
-                )
 
                 # init db manager
-                self.db_manager.open(passphrase=self.password)
+                error: str | None = self.db_manager.open(
```

Comment:
Please start by decrypting the private key and checking it is ok. If not, continue prompting for password (also add maximum number of retries, e.g. 3), if the key is decrypted fine, then assert the database is opened correctly. Do not return an error from db_manager.open() - it either succeeds with the given password (which you know to be valid from decrypting the key) or it throws an exception

Reply:
Done